packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -25,18 +30,37 @@ export * from './types';
 
 export const REQUEST_RATE = 10;
 
+if (process.env.REACT_APP_FAKE_CHANNEL_PROVIDER === 'true') {
```

Comment:
I think that the WebTorrentClient and the Web3TorrentChannelClient should be a little more independent of each other.
That way, Web3TorrentChannelClient could be another React Context Provider, and at the same time, be passed to the WebTorrentPaidStreamingClient as a constructor argument.

I know that this is a draft PR, so I don't want to judge too hard :-P

Reply:
That way, all these checks could be done in the Web3TorrentChannelClient, at that client creation.

Reply:
I'm working on getting this implemented 👍